### PR TITLE
fix: empty string check for aws url validation

### PR DESCRIPTION
**Pull request by sai-sunder-s**
Validate aws metadata urls only if they are not null or empty. It is possible to fetch tokens successfully even if metadata urls are not provided but required environment variables are set. If both metadata urls and env vars are not provided, we will throw an exception.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-auth-library-java/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

**Review by TimurSadykov — CHANGES_REQUESTED** [review]

File: oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java [hunk 1]
```diff
@@ -137,18 +137,21 @@ private void validateMetadataServerUrls() {
...
+    @VisibleForTesting
+    static void validateMetadataServerUrlIfAny(String urlString, String nameInConfig) {
+      if (urlString == null || urlString.trim().length() == 0) {
+        return;
+      }
+
+      try {
+        URL url = new URL(urlString);
+        String host = url.getHost();
+        if (!host.equals("169.254.169.254") && !host.equals("[fd00:ec2::254]")) {
```
Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java:149]: can host have different casing?

File: oauth2_http/javatests/com/google/auth/oauth2/AwsCredentialsTest.java [hunk 2]
```diff
@@ -489,6 +501,66 @@ public void getAwsSecurityCredentials_fromEnvironmentVariablesWithToken() throws
...
+    AwsSecurityCredentials credentials =
+        testAwsCredentials.getAwsSecurityCredentials(EMPTY_METADATA_HEADERS);
+
+    assertEquals("awsAccessKeyId", credentials.getAccessKeyId());
+    assertEquals("awsSecretAccessKey", credentials.getSecretAccessKey());
+    assertEquals("awsSessionToken", credentials.getToken());
+  }
+
+  @Test
+  public void validateMetadataServerUrlIfAny_validUrls() {
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/AwsCredentialsTest.java:528]: validOrEmptyUrls

File: oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java (hunk 1 above)
Comment (sai-sunder-s) [reply oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java:149]: Apparently both lower and upper case are acceptable but as per RFC, they should be lower case https://www.rfc-editor.org/rfc/rfc5952#section-4.3

**Review by TimurSadykov — APPROVED** [review]
LGTM, except i would allow uppercase host

**Pull request merged**